compute_endpoint/tests/integration/endpoint/executors/test_mpiengine.py

```diff
@@ -0,0 +1,89 @@
+import random
+import uuid
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_sdk.serialize import ComputeSerializer
+from tests.utils import ez_pack_function, get_env_vars
+
+temporary_skip = False
+try:
+    from globus_compute_sdk.sdk.bash_function import BashResult
+    from globus_compute_sdk.sdk.mpi_function import MPIFunction
+except ImportError:
+    temporary_skip = True
+
+
+@pytest.mark.skipif(temporary_skip, reason="Skip test until MPIFunctions are merged")
+def test_mpi_function(mpi_engine, tmp_path):
+    """Test for the right cmd being generated"""
+    engine, nodeslist = mpi_engine
+    queue = engine.results_passthrough
+    task_id = uuid.uuid1()
+    serializer = ComputeSerializer()
+
+    num_nodes = random.randint(1, len(nodeslist))
+    resource_spec = {"num_nodes": num_nodes, "num_ranks": random.randint(1, 4)}
+
+    mpi_func = MPIFunction("pwd", resource_specification=resource_spec)
+    task_body = ez_pack_function(
+        serializer, mpi_func, (), {"resource_specification": resource_spec}
+    )
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(task_id=task_id, task_buffer=task_body)
+    )
+    engine.submit(task_id, task_message, resource_specification=resource_spec)
+    flag = False
+    for _i in range(20):
+        q_msg = queue.get(timeout=10)
```

Comment:
Given that we control everything in the tests, is it possible to be more precise than "Up to 20 attempts with up to 10s wait for each iteration?"  I get that it will "never happen" in practice, but for tests, where we control the environment, this style feels very dirty.  I naively expect to see a single `.get()` call and expect the rest of the test to assert on the result of that single call.  If we truly need more attempts, then that implies to me that there is more work to understand the infrastructure.

Reply:
I've switched the logic here over to waiting on a future, which looks a lot cleaner.